### PR TITLE
Condition sets only persist if actively editing

### LINKED ISSUE #3954: Persistence errors when using Conditional Styling
We had seen this often on hulk, and now get lots of persistence errors when applying Conditional Styling on banner. This doesn't happen for me running locally:

image

### Reproduce
1. Create a Condition Set.
2. Create a Condition Widget.
3. Use the Condition Set to conditionally style the Condition Widget, then save changes.
4. Sporadically, but often, a persistence error will be displayed.

This also seems to occur when navving around in the tree, or viewing a Layout that includes a conditionally styled element.

### LINKED ISSUE #4146: Persistence errors in Condition Sets when editing Condition Widgets
#### Summary
Seen during Science Open MCT tagup 8-24-21. Persistence errors encountered frequently (but sporadically) and by multiple users when editing Condition Widgets that use Condition Sets for styling. Users were not able to save edited properties for the Condition Widget. Users were saving sets and widgets in their own independent folders. Trying to dupe this after the session, wasn't able to replicate.
Screen Shot 2021-08-25 at 7 34 55 AM

#### Expected vs Current Behavior
No errors.

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [X] Does this impact a critical component?
- [ ] Is this just a visual bug?

#### Steps to Reproduce

1. Make a Condition Set (CS) and a Condition Widget.
2. Use the CS to conditionally style the widget. Save it.
3. Edit properties for the widget, change the label and/or add a value into the URL field. Save.

#### Environment
* Open MCT Version: 1.7.7
* Deployment Type: /stable
* OS:
* Browser:

### PR DESCRIPTION
**Pull request by shefalijoshi**
Previously, the motivation to update a condition's description and save to the persistence store was that the condition description was shared by the condition set view as well as the Styles tab view (as we were trying to avoid calling object.get for telemetry endpoints in the Styles view)
Since this persistence is causing conflicts on couchDB when multiple users are using Open MCT, now we only save the summary for conditions if there is a user initiated operation.
Loading the composition of a condition set will still update the condition summaries to the condition set domain object but not save them to the persistence store. This should ensure that if telemetry endpoints go _missing_ after a condition set is created, the summary reflects this correctly.

Resolves #4146

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? Will users need to change how they are calling the API, or how they've extended core plugins such as Tables or Plots?

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/condition/ConditionManager.js [hunk 1]
```diff
@@ -139,10 +146,17 @@ export default class ConditionManager extends EventEmitter {
         }
     }
 
+    getConditionDescription(condition) {
+        condition.updateDescription();
```
Comment (nikhilmandlik) [inline src/plugins/condition/ConditionManager.js:150]: getConditionDescription is calling updateDescription on condition, feels incorrect. May be rename something else?

File: src/plugins/condition/ConditionManager.js [hunk 2]
```diff
@@ -220,7 +230,7 @@ export default class ConditionManager extends EventEmitter {
             this.conditionSetDomainObject.configuration.conditionCollection.unshift(newCondition);
         }
 
-        this.initCondition(newCondition, index);
+        this.initCondition(newCondition, index, true);
```
Comment (nikhilmandlik) [inline src/plugins/condition/ConditionManager.js:233]: may be can call 'init' and then 'update'?

File: src/plugins/condition/criterion/AllTelemetryCriterion.js [hunk 3]
```diff
@@ -177,7 +177,7 @@ export default class AllTelemetryCriterion extends TelemetryCriterion {
                 const timeSystem = this.openmct.time.timeSystem();
 
                 telemetryRequestsResults.forEach((results, index) => {
-                    const latestDatum = results.length ? results[results.length - 1] : {};
+                    const latestDatum = (results instanceof Array && results.length) ? results[results.length - 1] : {};
```
Comment (nikhilmandlik) [inline src/plugins/condition/criterion/AllTelemetryCriterion.js:180]: `Array.isArray(result)` ?

**Review by nikhilmandlik — COMMENTED** [review]
Looks good, Small changes/questions

File: src/plugins/condition/ConditionManager.js (hunk 2 above)
Comment (nikhilmandlik) [reply src/plugins/condition/ConditionManager.js:233]: hmm, ll it trigger mutate?

File: src/plugins/condition/criterion/AllTelemetryCriterion.js (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/condition/criterion/AllTelemetryCriterion.js:180]: Done

File: src/plugins/condition/ConditionManager.js (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/condition/ConditionManager.js]: no it won't

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Comment by nikhilmandlik** [comment]
### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)

**Pull request merged**